### PR TITLE
[DP-393] ORM Mapped entities extracted from Riak KV do not contain ke…

### PR DESCRIPTION
**Pull request by aleksey-suprun**
…y and index fields

### CODE REVIEW

File: connector/src/test/scala/com/basho/riak/spark/rdd/StoreRDDTest.scala [hunk 1]
```diff
@@ -210,15 +209,57 @@ class RDDStoreTest  extends AbstractRDDTest {
...
+        | ['${json-unit.ignore}', '["key1",1]'],
+        | ['${json-unit.ignore}', '["key2",2]'],
+        | ['${json-unit.ignore}', '["key3",3]']
+        | ]""".stripMargin, lData)
 
     verifyContentTypeEntireTheBucket("application/json", DEFAULT_NAMESPACE_4STORE)
   }
 
+  @Test
+  def readDataUsingCustomMapper(): Unit = {
```
Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/StoreRDDTest.scala]: Lets rename it to readDataUsingCustomConverterFactory

File: connector/src/test/scala/com/basho/riak/spark/rdd/StoreRDDTest.scala [hunk 2]
```diff
@@ -210,15 +209,57 @@ class RDDStoreTest  extends AbstractRDDTest {
...
+    val size = 4
+    val rdd = sc.parallelize((1 to size).map(_.toString))
+
+    implicit val vwf = new WriteDataMapperFactory[String, KeyValue] {
```
Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/StoreRDDTest.scala]: Can we get rid of Custom DM for better test locality

File: connector/src/main/scala/com/basho/riak/spark/util/RiakObjectConversionUtil.scala [hunk 3]
```diff
@@ -82,15 +60,11 @@ object RiakObjectConversionUtil {
...
 
-  private def parseContentTypeAndCharset(contentType: String): (String,String) ={
-    val d = contentType.split(";").transform(x=> x.trim.toLowerCase)
-
-    var charset = "UTF-8"
-    for( i <- 1 to d.length-1){
-      if(d(i).startsWith("charset")){
-        charset = d(i).substring(d(i).indexOf("=") + 1)
-      }
+  private def parseContentTypeAndCharset(contentType: String): (String, String) =
```
Comment (srgg) [inline connector/src/main/scala/com/basho/riak/spark/util/RiakObjectConversionUtil.scala]: It seems that this routine is never used..

File: connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala [hunk 4]
```diff
@@ -292,4 +301,31 @@ class ReadFromRiakRDDTest extends AbstractRDDTest{
...
     assertEquals(7, allValues.size)
   }
+
+  @Test
+  def readDataUsingCustomConverterFactory(): Unit = {
+    // The aim of this custom converter is to extract string timestamp values from UserTS objects and convert them
+    // to long values. In other words it converts UserTS.timestamp -> UserTsTimestamp
+    ConverterFactory.getInstance.registerConverterForClass(classOf[UserTSTimestamp], new Converter[UserTSTimestamp](classOf[UserTSTimestamp]) {
+      override def toDomain(binaryValue: BinaryValue, s: String): UserTSTimestamp = {
+        val jsonConverter = ConverterFactory.getInstance.getConverter(classOf[UserTS])
```
Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala]: Better to add comment kinda: "Using default Json conversion from java clietn to.. bla bla bla"

File: connector/src/main/scala/com/basho/riak/spark/util/RiakObjectConversionUtil.scala (hunk 3 above)
Comment (aleksey-suprun) [inline connector/src/main/scala/com/basho/riak/spark/util/RiakObjectConversionUtil.scala]: Removed

File: connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala [hunk 5]
```diff
@@ -17,29 +17,39 @@
...
+import com.basho.riak.client.api.convert.{Converter, ConverterFactory, JSONConverter}
+import com.basho.riak.client.core.util.BinaryValue
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
 import org.junit.Assert._
 import org.junit.Test
 import org.junit.experimental.categories.Category
 import com.basho.riak.spark._
 
 case class UserTS(timestamp: String, user_id: String)
+//class UserTSTimestamp(val timestamp: Long) extends Serializable
```
Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala]: Please remove this commented line

File: connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala [hunk 6]
```diff
@@ -17,29 +17,39 @@
...
 
 @Category(Array(classOf[IntegrationTests]))
 class ReadFromRiakRDDTest extends AbstractRDDTest{
+
+  JSONConverter.registerJacksonModule(DefaultScalaModule)
```
Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala]: I think this initialization should be somewhere in the Spark Connector already. Could you check it? From the users perspective it's better do not take care about such things..

File: connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala (hunk 5 above)
Comment (aleksey-suprun) [inline connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala]: Done

**Comment by srgg** [comment]
:+1:

**Pull request merged**